open-grant-proposals/orbitdb-pinning/README.md

```diff
@@ -0,0 +1,194 @@
+This Filecoin integration, and its future users are very likely to provide valuable
+context and feedback to the Filecoin Project, and drive the project roadmap forward.
+
+### Risks
+
+Of course this outcome is not guaranteed, and not without risk. During the initial planning
+process the grant team has identified the primary risks that could impact either the
+implementation process or the desired outcome of the grant effort.
+
+#### Last-minute changes/issues with Filecoin network/client behavior
+> Likelihood: High | Impact: Low
+
+As of this writing, Filecoin is still in active development prior to its planned MainNet launch.
+Things can and will change fast, potentially even after the MainNet launch. The grant team simply
+accepts this risk and we will plan accordingly. The practical upshot of this is that we will allot
+slightly more time in our milestone schedules to account for such contingencies.
+
+### Benefits
+
+If this project succeeds, OrbitDB's users will be happily pinning their data, and
+Filecoin will have a successful example of a net-zero, non-profit pinning community
+running within its network.
+
+Also, as mentioned in the [value](#value) section above, the effort itself will raise awareness of
+Filecoin as well as help steer the roadmap based on real-world usage.
+
+## Deliverables
+
+We plan on breaking this work up into two smaller milestones, described in the next section. At
+each milestone, we will deliver the following:
+
+### Milestone 1 - Planning and Implementation
+
+- A public repository with a working implementation of OrbitDB pinning using
+the Filecoin HTTP APIs
+- Documentation, tests, and instructions
+
+### Milestone 2 - Launch and Outreach
+
+- One or more blog posts detailing the need and solution presented
+- Social marketing efforts like posts and chat room announcements
+- A docker image or reference implementation
+
+## Development Roadmap
+
+As per the [Risks](#risks) section, this is subject to change as the project and Filecoin
+itself mature over the coming months.
+
+Note: The budget in the milestone is based on the _equivalent_ of one full-time developer working
+on the project at a time. This metric is used for ease of understanding and the work may be the
+result of many team members via more granular chunks of work.
+
+### Milestone 1 - Implementation
+
+The implementation will be split into its back-end and front-end components.
+
+#### Milestone 1a - Back-End Filecoin Network Implementation
+
+[`orbit-db-io`] is an existing package that handles reading and writing data from the
+IPFS DAG, and is the cornerstone of [`ipfs-log`]. The grant team would repeat this
+pattern to create `orbit-db-filecoin-io` which would be similarly composable with
+[`ipfs-log`].
+
+`orbit-db-filecoin-io` would follow Filecoin standards and utilize the Filecoin
+HTTP API, allowing it to interface with any compliant Filecoin nodes, including remotely
```

Comment:
Hi @aphelionz,

Filecoin nodes only have a JSON RPC API. An HTTP API would be available from a Filecoin-backed IPFS Pinning Service, the spec for which is being developed and some services that currently run IPFS Pinning APIs will run. 

Can you clarify if this module will integrate with the Filecoin lotus JSON RPC API or rely on a Pinning API?

Reply:
Good catch. Yes, we meant the JSON RPC API, not the pinning API. I will update the proposal.